Feature/scrum 9/view specific item

jjjjuuudde opened the pull request:
References SCRUM-9
Description: add view specific item
Change Type (feat, fix, etc.): feat

Code Review Checklist:

- [x] Tested feature manually via emulator
- [x] Created and passed all test cases
- [x] Consistent formatting
- [x] Clean code with no errors or warnings
- [x] Adheres to SOLID principles

jjjjuuudde committed:
added feat ViewSpecificItem and related xml, made Item.java parcelable. Not funtional since still need select Item implementation

jjjjuuudde committed:
Merge pull request #5 from David-Huynh/main

added feat ViewSpecificItem and related xml, made Item.java parcelable. Not funtional since still need select Item implementation

jjjjuuudde committed:
first commit didnt add ViewSpecificItem.java

jjjjuuudde committed:
added changes

jjjjuuudde committed:
modified item

jjjjuuudde committed:
test

jjjjuuudde committed:
fixed problem with Main Activity, messed up in last push

jjjjuuudde committed:
Item comment

jjjjuuudde committed:
updated ViewSpecificItem to work with CatalogItem

jjjjuuudde committed:
trying to pull new main

jjjjuuudde committed:
recreated ViewItem, added fragment to catalog_item_layout

jjjjuuudde committed:
Merge branch 'main' into feature/SCRUM-9/view-specific-item

jjjjuuudde committed:
catalog_item_layout.xml: added view button
CatalogItemAdapter: edited ViewHolder to include view button
ViewItem: added view item compatibility
fragment_catalog_view.xml: added fragmentContainerView
CatalogItem: implement Serializable

David-Huynh committed:
feat: onclick navigation and save to viewmodel

jjjjuuudde committed:
AndroidManifest.xml: added internet prems for image urls
fragment_catalog_view.xml: removed extra fragment container
ViewItemFragment.java: added async load image from url

jjjjuuudde committed:
updating branch, pull from main

David-Huynh commented on app/src/main/java/com/b07project2024/group1/Item.java:
Can you remove this file since its unused

David-Huynh commented on app/src/main/java/com/b07project2024/group1/MainActivity.java:
Remove these imports since they are unused now

David-Huynh commented on app/src/main/java/com/b07project2024/group1/CatalogItem.java:
```diff
@@ -1,15 +1,19 @@
  * The CatalogItem class is POJO that represents an item in the TAAM collection
  */
-public class CatalogItem {
+public class CatalogItem implements Serializable {
```
Can you remove this if you aren't using it?

David-Huynh commented on app/src/main/java/com/b07project2024/group1/CatalogFragment.java:
```diff
@@ -54,10 +58,14 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
+        catalogItemAdapter = new CatalogItemAdapter(this);
         recyclerView.setAdapter(catalogItemAdapter);
 
+//        loginViewModel = new ViewModelProvider(requireActivity()).get(LoginViewModel.class);
+
+
```
remove this

David-Huynh commented on app/src/main/java/com/b07project2024/group1/CatalogFragment.java:
```diff
@@ -1,5 +1,6 @@
 package com.b07project2024.group1;
 
+import android.content.ContextWrapper;
```
remove unused imports

David-Huynh reviewed (changes requested):
There's also some weird formatting issues with the view for me the text overlaps and i dont see the pictures
![image](https://github.com/user-attachments/assets/c9d73a6a-e299-497d-b973-a28990726f7c)

jjjjuuudde committed:
fix
AndroidManifest.xml: added ACCESS_NETWORK_STATE perms
CatalogFragment.java: removed unused lines
CatalogItem.java: removed Serializable
fragment_view_item.xml: fixed text overlap
MainActivity.java: removed unused lines
ViewItemFragment.java: changed Bitmap to Glide

jjjjuuudde committed:
fix
AndroidManifest.xml: added ACCESS_NETWORK_STATE perms
CatalogFragment.java: removed unused lines
CatalogItem.java: removed Serializable
fragment_view_item.xml: fixed text overlap
MainActivity.java: removed unused lines
ViewItemFragment.java: changed Bitmap to Glide
